### ISSUE TITLE
Autofocus

### ISSUE THREAD

**Issue by mrlarssen** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#72]
Is there a way to enable autofocus on android and ios ?

**Comment by RichardBoyder** [comment]
Any reply to this ?

**Comment by YouCantTriforce** [comment]
I'd love to be able to focus on touch. Is it ever gonna be possible?

**Comment by YouCantTriforce** [comment]
Okay, that was easy for Android part and I am going to give you quick implementation of that.
Probably it should check if preview is running for camera but I'm serving this only as example.
Go to platforms/android/platform_www/plugins/com.mbppower.camerapreview/www/CameraPreview.js and add following code
```
CameraPreview.focus = function() {
  exec(null, null, PLUGIN_NAME, "focus", []);
};
```
Then in platforms/android/src/com/mbppower/CameraPreview.java add
```
	private boolean focusCamera(final JSONArray args, CallbackContext callbackContext) {
		if(fragment == null){
			return false;
		}
		fragment.refocusCamera();
		return true;
	}
```
after that in platforms/android/src/com/mbppower/CameraActivity.java add
```
		public void refocusCamera()
		{
				getActivity().runOnUiThread(new Runnable() {
					@Override
					public void run() {
						mCamera.cancelAutoFocus();
						mCamera.autoFocus(new Camera.AutoFocusCallback() {
								@Override
								public void onAutoFocus(boolean success, Camera camera) {}
						});
					}
				});
		}
```
Then you can use it from javascript in your application, e.g.
```
		document.getElementById('focus_button').addEventListener("click", function(){
			cordova.plugins.camerapreview.focus();
		});
```

**Comment by andrea-86** [comment]
+1
nice solution!

**Comment by andrea-86** [comment]
anything for IOs device?

**Comment by ptrpt** [comment]
@YouCantTriforce Thanks a lot for the question and the solution. Can't we add it to the project itself? It would prevent people of doing this fix, over and over.

**Comment by YouCantTriforce** [comment]
I can only provide code for android as I'm not familiar with ios development. Since this project is meant to be crossplatform I did not create pull request.

**Comment by ptrpt** [comment]
@YouCantTriforce I thought that the issue with the autofocus was only on the Android-side of the plugin. My bad ;) Thanks by the way!

**Comment by YouCantTriforce** [comment]
It's not actually an issue. It is missing feature.

**Comment by erperejildo** [comment]
Such a good solution!
Why is not implemented in latest versions? Sometimes we have to remove platforms losing these changes

**Comment by psiciliano** [comment]
Hi! Nice solution! Sorry if is a newbie question, but I'm trying to implement it. I've changed com.mbppower.camerapreview\src\android\com\mbppower\CameraActivity.java, com.mbppower.camerapreview\src\android\com\mbppower\CameraPreview.java and com.mbppower.camerapreview\www\CameraPreview.js files, but when I've tried to do cordova run android --device it not recompiles with the changes.
I've tried to do cordova clean but it doesn't work either. How did I miss to do the patch.
